lib/ansible/modules/source_control/git.py

```diff
@@ -432,6 +439,21 @@ def clone(git_path, module, repo, dest, remote, depth, version, bare,
                         "HEAD, branches, tags or in combination with refspec.")
     if reference:
         cmd.extend(['--reference', str(reference)])
+    separate_git_dir_fallback = False
+    if separate_git_dir is not None and bare:
+        module.warn("Ignoring separate_git_dir argument. "
+                    "Can not do bare clone with argument separate_git_dir.")
+    elif separate_git_dir:
+        git_version_used = git_version(git_path, module)
+        if git_version_used is None:
+            raise AnsibleError("Can not find git executable.")
+        if git_version_used < LooseVersion('1.7.5'):
+            # git before 1.7.5 doesn't have separate-git-dir argument, do fallback
+            separate_git_dir_fallback = True
```

Comment:
You should not postpone a fallback execution. It should happen where clone happens so that all subsequent commands/checks would operate on the final version of workdir + repo.

Reply:
yes, I will move fallback code right after the clone command.